WordPress/src/main/java/org/wordpress/android/util/LiveDataUtil.kt

```diff
@@ -0,0 +1,20 @@
+package org.wordpress.android.util
+
+import android.arch.lifecycle.LifecycleOwner
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.Observer
+
+/**
+ * Use this in order to observe an emission only once - for example for displaying a Snackbar message
+ */
+fun <T : Event> LiveData<T>.observeEvent(owner: LifecycleOwner, observer: (T) -> Boolean) {
+    this.observe(owner, Observer {
+        if (it != null && !it.hasBeenHandled) {
+            if (observer(it)) {
+                it.hasBeenHandled = true
+            }
+        }
+    })
+}
+
+open class Event(var hasBeenHandled: Boolean = false)
```

Comment:
We already talked about this on the call, but just adding a note that if we end up keeping the code, it'd be good to improve this. I think we could make this an interface, but if it has to be a class, let's at least make it `abstract`. Also, let's not expose the `hasBeenHandled` and instead just have a method such as `handle` and keep the state internal.

Reply:
I've removed it. However, even if we decide to keep it after all, I'd suggest improving it in another PR as it would create merge conflicts with Vojta's branch.

Reply:
Yeap, that makes sense. If that turns out to be the case, do you mind opening an issue for that so we don't forget about it?